src/test/kotlin/MicronautAuthenticationTest.kt

```diff
@@ -0,0 +1,246 @@
+    }
+
+    @Test
+    fun `ensure that global security and operational security defined will set the correct parameters `() {
+        val controller = setupTest("operation_authentication_with_global.yml")
+
+        val prohibitedController = (controller.find({it.name == "ProhibitedController"})!!.members.single() as TypeSpec).funSpecs.single()
+        val requiredController = (controller.find({it.name == "RequiredController"})!!.members.single() as TypeSpec).funSpecs.single()
+        val optionalController = (controller.find({it.name == "OptionalController"})!!.members.single() as TypeSpec).funSpecs.single()
+        val defaultController = (controller.find({it.name == "DefaultController"})!!.members.single() as TypeSpec).funSpecs.single()
+        val noneController = (controller.find({it.name == "NoneController"})!!.members.single() as TypeSpec).funSpecs.single()
+
+        assertThat(prohibitedController.annotations).anySatisfy{member ->
+            assertThat(member.members).anyMatch{it.toString() == "SecurityRule.IS_ANONYMOUS"}}
+        assertThat(requiredController.annotations).anySatisfy{member ->
+            assertThat(member.members).anyMatch{it.toString() == "SecurityRule.IS_AUTHENTICATED"}}
+        assertThat(optionalController.annotations).anySatisfy{member ->
+            assertThat(member.members).anyMatch{it.toString() == "SecurityRule.IS_AUTHENTICATED, SecurityRule.IS_ANONYMOUS"}}
+        // expect that explicitly set to no authentication
+        assertThat(noneController.annotations).noneSatisfy{member ->
+            assertThat(member.members).anyMatch{it.toString() == "SecurityRule.IS_AUTHENTICATED"}
+            assertThat(member.members).anyMatch{it.toString() == "SecurityRule.IS_ANONYMOUS"}
+            assertThat(member.members).anyMatch{it.toString() == "SecurityRule.IS_AUTHENTICATED, SecurityRule.IS_ANONYMOUS"}}
+        // expect IS_AUTHENTICATED because of the global setting
+        assertThat(defaultController.annotations).anySatisfy{member ->
+            assertThat(member.members).anyMatch{it.toString() == "SecurityRule.IS_AUTHENTICATED"}}
+
+
+        assertThat(prohibitedController.parameters).noneSatisfy{member ->
+            assertThat(member.name).isEqualTo("authentication")}
+        assertThat(requiredController.parameters).anySatisfy{member ->
+            assertThat(member.name).isEqualTo("authentication")}
+        assertThat(optionalController.parameters).anySatisfy{member ->
+            assertThat(member.name).isEqualTo("authentication")}
+        assertThat(noneController.parameters).noneSatisfy{member ->
+            assertThat(member.name).isEqualTo("authentication")}
+        // expect authenticated because of the global setting
+        assertThat(defaultController.parameters).anySatisfy{member ->
+            assertThat(member.name).isEqualTo("authentication")}
+
+    }
+
+    @Test
+    fun `ensure that ONLY operational security defined will set the correct parameters `() {
```

Comment:
**suggestion:** `ensure that only operation-level security being defined will result in the correctly generated code`